google-cloud-core/src/main/java/com/google/cloud/Identity.java

```diff
@@ -280,10 +280,12 @@ public String strValue() {
   public static Identity valueOf(String identityStr) {
     String[] info = identityStr.split(":");
     Type type = Type.valueOf(CaseFormat.LOWER_CAMEL.to(CaseFormat.UPPER_UNDERSCORE, info[0]));
-    if (type.equals(Type.ALL_AUTHENTICATED_USERS) || type.equals(Type.ALL_USERS)) {
+    if (info.length == 1) {
```

Comment:
This allows someone to pass in `ALL_AUTHENTICATED_USERS` with an argument, or something other than the `ALL_*` types without an argument, and still get a result.